applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_linear_elastic_plane_strain_2D_law.cpp

```diff
@@ -0,0 +1,201 @@
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+
+    ConstitutiveLaw::Features law_features;
+    law.GetLawFeatures(law_features);
+
+    KRATOS_EXPECT_TRUE(law_features.mOptions.IsDefined(ConstitutiveLaw::PLANE_STRAIN_LAW))
+    KRATOS_EXPECT_TRUE(law_features.mOptions.IsDefined(ConstitutiveLaw::INFINITESIMAL_STRAINS))
+    KRATOS_EXPECT_TRUE(law_features.mOptions.IsDefined(ConstitutiveLaw::ISOTROPIC))
+
+    const auto& strain_measures = law_features.mStrainMeasures;
+    KRATOS_EXPECT_TRUE(std::find(strain_measures.begin(), strain_measures.end(),
+                                 ConstitutiveLaw::StrainMeasure_Infinitesimal) != strain_measures.end())
+    KRATOS_EXPECT_TRUE(std::find(strain_measures.begin(), strain_measures.end(),
+                                 ConstitutiveLaw::StrainMeasure_Deformation_Gradient) !=
+                       strain_measures.end())
+
+    KRATOS_EXPECT_EQ(law_features.mStrainSize, 4);
+    KRATOS_EXPECT_EQ(law_features.mSpaceDimension, 2);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawReturnsExpectedStrainSize, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+    KRATOS_EXPECT_EQ(law.GetStrainSize(), 4);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawReturnsExpectedWorkingSpaceDimension,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+    KRATOS_EXPECT_EQ(law.WorkingSpaceDimension(), 2);
+}
+
+Vector CalculateStrain(GeoLinearElasticPlaneStrain2DLaw& rConstitutiveLaw)
+{
+    ConstitutiveLaw::Parameters parameters;
+    parameters.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN);
+
+    Vector strain;
+    parameters.SetStrainVector(strain);
+
+    rConstitutiveLaw.CalculateMaterialResponsePK2(parameters);
+
+    return strain;
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawReturnsExpectedStress, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+
+    const auto stress = CalculateStress(law);
+
+    Vector expected_stress{4};
+    expected_stress <<= 2.5e+07, 2.5e+07, 2.5e+07, 3.84615e+06;
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(expected_stress, stress, 1e-3);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawReturnsExpectedStress_WhenOnlyDiagonalEntriesAreConsidered,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+    law.SetConsiderDiagonalEntriesOnlyAndNoShear(true);
+
+    const auto stress = CalculateStress(law);
+
+    Vector expected_stress{4};
+    expected_stress <<= 1.34615e+07, 1.34615e+07, 1.34615e+07, 0;
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(expected_stress, stress, 1e-3);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawReturnsExpectedStress_WithInitialStressAndStrain,
```

Comment:
When I read the test case name, I had expected that we would use member `SetInitialState` of the constitutive law. That works a bit different here. Out of curiosity: what is the expected effect of using `SetInitialState`?

Reply:
I don't think it's implemented for this law. I think it's a good idea to change that at some point and use that method instead of the one that's now available (shown in this UT)